modules/apps/frontend-js/frontend-js-alert-support-web/src/main/java/com/liferay/frontend/js/alert/support/web/internal/servlet/taglib/AlertBottomDynamicInclude.java

```diff
@@ -38,16 +37,9 @@ public void include(
 			HttpServletResponse httpServletResponse, String key)
 		throws IOException {
 
-		ScriptData scriptData = new ScriptData();
-
-		String initModuleName = _npmResolver.resolveModuleName(
-			"frontend-js-alert-support-web/index");
-
-		scriptData.append(
-			null, "AlertProvider.default()",
-			initModuleName + " as AlertProvider", ScriptData.ModulesType.ES6);
-
-		scriptData.writeTo(httpServletResponse.getWriter());
+		_jsModuleLauncher.writeScript(
+			httpServletResponse.getWriter(), "frontend-js-alert-support-web",
+			"AlertProvider", "AlertProvider.default()");
```

Comment:
Still trying to wrap my head around this a little...

- `frontend-js-alert-support-web` is the name of the module, as per its `package.json` and Webpack is compiling its `index.js` as a main entry point?
- How's the `AlertProvider` variable/module being generated? What are the contents of such function?

Reply:
- `"frontend-js-alert-support-web"` is the name of the module
- `"AlertProvider"` is the name of the variable to which we will assign the module (it could be anything else, like `perico`, for example).
- `AlertProvider.default()` is relying on the previous variable name (so this could be `perico.default()` if we decided to change the var name).

Reply:
It's like ' Liferay.Loader.require`...

Reply:
```javascript
Liferay.Loader.require(
  "frontend-js-alert-support-web", 
  function(AlertProvider) {
    AlertProvider.default();
  }
);